CMakePresets.json

```diff
@@ -13,10 +13,10 @@
       "generator": "Unix Makefiles",
       "binaryDir": "${sourceDir}/out/build/${presetName}",
       "environment": { "CMAKE_BUILD_PARALLEL_LEVEL": "8" },
+      "toolchainFile": "${sourceDir}/out/default/build/Debug/generators/conan_toolchain.cmake",
```

Comment:
Maybe you want to try the approach of including Conan generated presets? Not compatible with ``cmake-conan`` integration with new CMake-providers, but the toolchain is mostly redundant with that integration (for the consumer side, it is still necessary for the package create)

Reply:
This is what I was asking already here:
> @prince-chrismc
> 
> Unfortunately we can't use conan-cmake (develop2), because we use our own CMakePresets.json file: Including both the conan provider `CMAKE_PROJECT_TOP_LEVEL_INCLUDES=${sourceDir}/conan-cmake/conan_provider.cmake` and the conan toolchain `CMAKE_TOOLCHAIN_FILE=${sourceDir}/out/default/build/Debug/generators/conan_toolchain.cmake` is not possible, because then cmake is complaining about the missing toolchain file which should be generated by conan.
> 
> So `conan install` is the only way for now?!

The problem is that we have different target architectures in our presets and as far as I could read from the docs this is not supported/will not be supported.

I'm still trying to figure out, whats the best approach to trigger conan install automatically? Probably I'll write our own small "cmake-wrapper" for that. The chicken-egg problem is, as you mention everywhere, that we have to know for which profile we want to call conan install.